test/__init__.py

```
@@ -19,6 +26,75 @@
+    sys.path.insert(0, str(target_dir))
+    if marker.exists():
+        return
+
+    cache_root.mkdir(parents=True, exist_ok=True)
+    with open(cache_root / f"{target_dir.name}.lock", "w") as lock_file:
+        fcntl.flock(lock_file, fcntl.LOCK_EX)
+        try:
+            if marker.exists():  # another process installed while we were waiting for the lock
+                return
+            target_dir.mkdir(parents=True, exist_ok=True)
+            subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)